src/treetensornetworks/solvers/alternating_update.jl

```diff
@@ -46,55 +46,60 @@ function alternating_update(
   (sweep_observer!)=observer(),
   sweep_printer=sweep_printer,
   write_when_maxdim_exceeds::Union{Int,Nothing}=nothing,
+  updater_kwargs,
   kwargs...,
 )
   maxdim, mindim, cutoff, noise, kwargs = process_sweeps(nsweeps; kwargs...)
 
   psi = copy(psi0)
 
-  insert_function!(sweep_observer!, "sweep_printer" => sweep_printer)
+  insert_function!(sweep_observer!, "sweep_printer" => sweep_printer) # FIX THIS
 
-  for sweep in 1:nsweeps
-    if !isnothing(write_when_maxdim_exceeds) && maxdim[sweep] > write_when_maxdim_exceeds
+  for which_sweep in 1:nsweeps
+    if !isnothing(write_when_maxdim_exceeds) &&
+      maxdim[which_sweep] > write_when_maxdim_exceeds
       if outputlevel >= 2
         println(
-          "write_when_maxdim_exceeds = $write_when_maxdim_exceeds and maxdim[sweep] = $(maxdim[sweep]), writing environment tensors to disk",
+          "write_when_maxdim_exceeds = $write_when_maxdim_exceeds and maxdim[which_sweep] = $(maxdim[which_sweep]), writing environment tensors to disk",
         )
       end
       PH = disk(PH)
     end
 
     sw_time = @elapsed begin
-      psi, PH = update_step(
-        solver,
+      psi, PH = sweep_update(
+        updater,
         PH,
         psi;
         outputlevel,
-        sweep,
-        maxdim=maxdim[sweep],
-        mindim=mindim[sweep],
-        cutoff=cutoff[sweep],
-        noise=noise[sweep],
+        which_sweep,
+        sweep_params=(;
+          maxdim=maxdim[which_sweep],
+          mindim=mindim[which_sweep],
+          cutoff=cutoff[which_sweep],
+          noise=noise[which_sweep],
+        ),
```

Comment:
I would define this up above before the call to `sweep_update` in a `NamedTuple` called `sweep_params`, and then pass that `sweep_params` as a keyword argument, to make this function call simpler.

Also, `sweep_params` doesn't seem like a good name for this, I think we decided on a different name for it but I forget what it was. Maybe these can be bundled together in the `updater_kwargs`, since they are just another set of arguments being passed to the updater?

Reply:
They are also required inside `region_update` (specifically inside `extract/insert local_tensor`). We could decide to move them into updater_kwargs, and then extract them when needed.
With a plan to move to allow to specify each function in region_update (extract, updater, insert) and it's kwargs via NamedTuple attached to the region, I'd for now avoid to putting them all into `updater_kwargs`.